### PR TITLE
add day 12

### PR DESCRIPTION
**Pull request by arora-aditya**

### CODE REVIEW

File: 2020/day-12/arora-aditya/day12.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    dir, count = line[0], int(line[1:])
+    if dir in set("NEWS"):
+        de, dn = process(dir, count)
+        we += de
+        wn += dn
+    elif dir == "F":
+        de, dn = process(F, count)
+        for i in range(count):
+            e += we
+            n += wn
```
Comment (arora-aditya) [inline 2020/day-12/arora-aditya/day12.py:90]: memes that I didn't multiply here lmao

Comment (AngelOnFira) [reply 2020/day-12/arora-aditya/day12.py:90]: lmao I was about to do that too, but thought about it for an extra second

File: 2020/day-12/arora-aditya/day12.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    elif dir == "L":
+        if count % 90 != 0:
+            sys.exit(1)
+        for i in range(ceil(count/90)):
+            F = order[order.index(F)-1]
+    elif dir == "R":
+        if count % 90 != 0:
+            sys.exit(1)
+        for i in range(ceil(count/90)):
+            F = order[(order.index(F)+1)%len(order)]
```
Comment (AngelOnFira) [inline 2020/day-12/arora-aditya/day12.py:54]: damn nice solution

Another student here did this, which took me a sec to get, but is actually pretty nifty

```python
def rotate_waypoint(x, y, r):
    count = r // 90
    for _ in range(count):
        x, y = y, x
        x *= -1
    return x, y
```

Comment (arora-aditya) [reply 2020/day-12/arora-aditya/day12.py:54]: that is indeed nifty

**Pull request merged**